### ISSUE TITLE
Enhancement: Linux Flatpak build

### ISSUE THREAD

**Issue by fpiesche** [FunkinCrew/Funkin#3057]
### Issue Checklist

- [X] I have properly named the enhancement
- [X] I have checked the issues/discussions pages to see if the enhancement has been previously suggested

### What is your suggestion, and why should it be implemented?

At a friend's request I've recently set up a Flatpak Linux build for [PsychEngine](https://flathub.org/apps/io.github.shadowmario.fnf-psychengine), a fork of Funkin with mod support (before Funkin itself got updated for this).

As the upstream variant of PsychEngine and the "official" release of FNF, I think it might be worthwhile getting Funkin' distributed as a Flatpak as well - this would be especially useful for e.g. Steam Deck users, where Flatpak is the main path for getting software that isn't on Steam.

I'd be happy to repurpose my Flatpak build setup for PsychEngine to build an official Funkin' release instead; as an added bonus Flathub's CI system would monitor external components such as Haxe, Flixel, etc for updates and automatically run test builds as those dependencies update - this would be helpful to Funkin' upstream as an early warning system for updates to haxelib dependencies breaking Funkin' itself (as well as to show when dependencies update that !

From your end, maintenance effort would be minimal; the Flathub build setup will basically just pull the sources from this GitHub repo, create a Linux build and publish it to flathub.org (as seen with PsychEngine above) for users to install and use.

**Comment by fpiesche** [comment]
For reference, [here](https://github.com/flathub/io.github.shadowmario.fnf-psychengine/tree/master)'s what a Flatpak build manifest looks like. The main requirements are [FreeDesktop metadata](https://freedesktop.org/software/appstream/docs/chap-Metadata.html) and an icon image in the upstream repository (if possible); the actual build pipeline is specified in [this yaml file](https://github.com/flathub/io.github.shadowmario.fnf-psychengine/blob/master/io.github.shadowmario.fnf-psychengine.yml) with the build dependencies for haxelib in [this supplementary file](https://github.com/flathub/io.github.shadowmario.fnf-psychengine/blob/master/psychengine-haxelib-dependencies.yml) for readability.

**Comment by Zombiepunk10** [comment]
This would be awesome actually, not only would it help with easy of download, but it would make it so the game could auto update! i think... I would prefer an official version though, not anything 3rd-party like the links above (although i might check them out either way

although, it might also mess up modding ease, so this version might have to have something that tells us were the mod folder is placed 😅

**Comment by Kasumicic** [comment]
Just wanted to add my voice to this. As an Ubuntu 22.04 LTS user, I can confirm that the current `.zip` build is completely unusable due to the glibc version requirement.

Finding this game unplayable on one of the most popular Linux distributions is really frustrating. A Flatpak would solve this issue for everyone instantly.

Really hope this gets considered soon. Thanks!

**Comment by Pucas01** [comment]
Hey guys,

I'm the maintainer of the project linked above and want to let everyone know I'm still working on it!

Yes Mods fully work (and the directory is listed on the GitHub)

My flatpak also includes SDL3 libs to allow the use of Wayland.

I also wanted to say i just pushed an updated that "fixes" #5311 by incorporating the script that DamianBossPL made INTO the flatpak, i also had to add a second wrapper script to fix a Wayland specific issue, which has not come up yet in the normal Linux version as it does not use Wayland.

I'm still hoping for an official flatpak release of the game, or allowing me to upload this version to flathub :) 

Until that happens expect updates on the same day as the Linux build releasing!

**Comment by Hundrec** [comment]
Hey, thanks for updating us on this! I'll see what Eric thinks of it!

**Comment by Kade-github** [comment]
> Just wanted to add my voice to this. As an Ubuntu 22.04 LTS user, I can confirm that the current `.zip` build is completely unusable due to the glibc version requirement.
> 
> Finding this game unplayable on one of the most popular Linux distributions is really frustrating. A Flatpak would solve this issue for everyone instantly.
> 
> Really hope this gets considered soon. Thanks!

Hello, I build the linux builds. This issue has been resolved I believe, as I was able to run Funkin on Ubuntu 24.04.3 LTS using the build I generated for v0.8.1. Only having to install libvlc-dev.

**Comment by Kade-github** [comment]
Secondly, I think flatpak would get outdated for us very quickly. Not as in, the version of the game. But the fact that our distributor of files would eventually (and hopefully) become steam. 

Kind of what I am saying, is that: We already have a pretty good way of distributing builds, and it's the same way every other OS gets them. This is kind of like asking us to put Funkin on the chocolatey user repository for Windows, which we would never really do. Not that we don't want too, it's just not in our scope.

Basically, atm with our current plans. Having the game on Flatpak and other derivatives is not something we are planning to do.
